### PR TITLE
Samples for BoxDecoration and some related classes

### PR DESCRIPTION
**Pull request by Hixie**

### CODE REVIEW

**Comment by Hixie** [comment]
@InMatrix, @sethladd, do you have any comments on how this looks?

File: packages/flutter/lib/src/widgets/container.dart [hunk 1]
```diff
@@ -16,10 +16,34 @@ import 'image.dart';
 /// not.
 ///
 /// Commonly used with [BoxDecoration].
-/// 
+///
+/// ## Sample usage
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/container.dart]: Why sample usage rather than sample code?

**Comment by abarth** [comment]
I wonder if we should build something similar to what we have on the web site to pull out these samples and analyze them to make sure they stay up-to-date.

File: packages/flutter/lib/src/widgets/container.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/container.dart]: I'm fine either way.

**Comment by Hixie** [comment]
> I wonder if we should build something similar to what we have on the web site to pull out these samples and analyze them to make sure they stay up-to-date.

Yes, we should definitely do that. I wanted to write a few of these to figure out what kind of patterns emerged before trying. In particular, it seems like I sometimes want to write code, sometimes expressions with trailing commas, and sometimes complete classes, so far.

**Comment by Hixie** [comment]
(I thought it might be that I wanted to write fake code with `...` placeholders and such, but that hasn't been the case so far.)

File: packages/flutter/lib/src/widgets/container.dart (hunk 1 above)
Comment (abarth) [inline packages/flutter/lib/src/widgets/container.dart]: I just meant that you mixed the two headings in this patch.

**Comment by abarth** [comment]
LGTM

File: packages/flutter/lib/src/widgets/container.dart (hunk 1 above)
Comment (Hixie) [inline packages/flutter/lib/src/widgets/container.dart]: Ah, my bad. Will fix.

**Comment by InMatrix** [comment]
Re: patterns

One of the primary use cases these snippets need to support is "copy & paste." So I'd avoid using `...` placeholders. If we really need to use them to make the snippet short, we should style them in such a way that it's really obvious the code is incomplete.

**Comment by Hixie** [comment]
@InMatrix That makes sense to me and is consistent with my experience so far.

**Pull request merged**